File: java-api/src/main/java/com/webmvc/javaapi/domain/Address.java
package com.webmvc.javaapi.domain;

import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Address {

    private String city;
    private String street;
    private String zipcode;

    public static Address create(String city, String street, String zipcode) {
        Address address = new Address();

        address.city = city;
        address.street = street;
        address.zipcode = zipcode;

        return address;
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/Address.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.Embeddable

@Embeddable
class Address protected constructor(
    val city: String,
    val street: String,
    val zipcode: String
) {
    companion object {
        fun create(city: String, street: String, zipcode: String) =
            Address(
                city = city,
                street = street,
                zipcode = zipcode
            )
    }
}